cmsplugin_filer_image/migrations/0002_delete_duplicate_thumbnailoptions.py

```diff
@@ -0,0 +1,30 @@
+from django.db import migrations
+from django.db.models import Count, Max
+
+
+def forward(apps, schema_editor):
+    """
+    Get all the duplicates for ThumbnailOption.
+    Point each FilerImage where the duplicates are used to the correct
+    ThumbnailOption.
+    Delete remaining ThumbnailOption duplicates.
+    """
+    ThumbnailOption = apps.get_model("cmsplugin_filer_image", "ThumbnailOption")
+    unique_fields = ["name", "width", "height", "crop", "upscale"]
+    duplicates = (ThumbnailOption.objects.values(*unique_fields)
+                  .order_by()
+                  .annotate(max_id=Max('id'), count_id=Count('id'))
+                  .filter(count_id__gt=1))
+
+    for duplicate in duplicates:
+        for thumbnail_option in ThumbnailOption.objects.filter(
+                **{x: duplicate[x] for x in unique_fields}
+        ).exclude(id=duplicate['max_id']):
```

Comment:
I think you should save this queryset in a variable outside the for statement. It's much more readable(by dropping the weird round closing bracket in the for statement) and it's better for debugging purposes(think of the case where you might need to put a pdb to check the result of the queryset).